manifests/init.pp

```diff
@@ -219,6 +223,34 @@
     mode    => '0600',
   }
 
+  # if given a jwk base64 binary file, we will use it
+  # otherwise we will auto generate one
+  if $jwk_key_store_file != '' {
+    file { $jwk_key_store_path:
+      ensure  => 'present',
+      content => base64('decode', $jwk_key_store_file),
+      path    => $jwk_key_store_path,
+      require => File['/etc/irida'],
+      notify  => Service['tomcat'],
+      owner   => $tomcat_user,
+      group   => $tomcat_group,
+      mode    => '0600',
+    }
+  }
+  else {
+    exec { 'Auto-generate JWK Key store file':
+      command  => "keytool -genkeypair -alias JWK -keyalg RSA -noprompt -dname 'CN=${server_base_url}, OU=ID, O=IRIDA, L=IRIDA,S=IRIDA,\
```

Comment:
You should set the sensitive variables as environment variables and then reference them that way. Otherwise you are potentially exposing them to anyone who can look at the process list